### ISSUE TITLE
Microsoft .NET exception

### ISSUE THREAD

**Issue by KittQB** [Angelelz/WinHelloUnlock#44]
**Describe the bug**
During the first configuration I've got an exception, just after I've scan my fingerprfint in "Hello".

**Screenshots**
![image](https://user-images.githubusercontent.com/61204294/74764631-5ac04e80-5282-11ea-989b-9def31282dfb.png)

**Additional context**
************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at WinHelloUnlock.Library.GetKeys(PwDatabase db)
   at WinHelloUnlock.UWPLibrary.<CreateHelloData>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at WinHelloUnlock.WinHelloUnlockExt.<FileOpenedHandler>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4075.0 built by: NET48REL1LAST
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
KeePass
    Assembly Version: 2.44.0.19625
    Win32 Version: 2.44.0.0
    CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/KeePass.exe
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4001.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4042.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4110.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
WinHelloUnlock
    Assembly Version: 1.5.0.0
    Win32 Version: 1.5
    CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/Plugins/WinHelloUnlock.dll
----------------------------------------
Windows.Security
    Assembly Version: 255.255.255.255
    Win32 Version: 10.0.10011.16384
    CodeBase: file:///C:/windows/system32/WinMetadata/Windows.Security.winmd
----------------------------------------
Windows.Storage
    Assembly Version: 255.255.255.255
    Win32 Version: 10.0.10011.16384
    CodeBase: file:///C:/windows/system32/WinMetadata/Windows.Storage.winmd
----------------------------------------
Windows.Foundation
    Assembly Version: 255.255.255.255
    Win32 Version: 10.0.10011.16384
    CodeBase: file:///C:/windows/system32/WinMetadata/Windows.Foundation.winmd
----------------------------------------
System.Runtime.WindowsRuntime
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.WindowsRuntime/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.WindowsRuntime.dll
----------------------------------------
KeeCloud
    Assembly Version: 1.2.1.11
    Win32 Version: 1.2.1.11
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/FNo8l5jIHUo0FXXbvDeh/KeeCloud.dll
----------------------------------------
KeePassRPC
    Assembly Version: 2.0.34.27662
    Win32 Version: 1.10.0.0
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/oyh8tdDgPdCetiZ74KmP/KeePassRPC.dll
----------------------------------------
Fleck2
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/oyh8tdDgPdCetiZ74KmP/Fleck2.DLL
----------------------------------------
DomainPublicSuffix
    Assembly Version: 2.0.1.0
    Win32 Version: 2.0.1.0
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/oyh8tdDgPdCetiZ74KmP/DomainPublicSuffix.DLL
----------------------------------------
Jayrock
    Assembly Version: 0.9.12915.0
    Win32 Version: 0.9.11214.1056
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/oyh8tdDgPdCetiZ74KmP/Jayrock.DLL
----------------------------------------
KeePass.XmlSerializers
    Assembly Version: 2.44.0.19625
    Win32 Version: 2.44.0.19625
    CodeBase: file:///C:/Program%20Files%20(x86)/KeePass%20Password%20Safe%202/KeePass.XmlSerializers.DLL
----------------------------------------
Jayrock.Json
    Assembly Version: 0.9.12915.0
    Win32 Version: 0.9.13422.2206
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/oyh8tdDgPdCetiZ74KmP/Jayrock.Json.DLL
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Security
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Security/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Security.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
Dropbox.Api
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.6472
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/FNo8l5jIHUo0FXXbvDeh/Dropbox.Api.DLL
----------------------------------------
System.Net.Http
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll
----------------------------------------
System.Net.Http.WebRequest
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http.WebRequest/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.WebRequest.dll
----------------------------------------
System.Threading.Tasks
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Threading.Tasks/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Threading.Tasks.dll
----------------------------------------
Microsoft.Threading.Tasks
    Assembly Version: 1.0.12.0
    Win32 Version: 1.0.168.0
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/FNo8l5jIHUo0FXXbvDeh/Microsoft.Threading.Tasks.DLL
----------------------------------------
Newtonsoft.Json
    Assembly Version: 7.0.0.0
    Win32 Version: 7.0.1.18622
    CodeBase: file:///C:/Users/b6c/AppData/Local/KeePass/PluginCache/FNo8l5jIHUo0FXXbvDeh/Newtonsoft.Json.DLL
----------------------------------------
System.Net
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Net/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.dll
----------------------------------------

**Comment by Angelelz** [comment]
Thank you for taking the time to report the issue. Could you please confirm if the option "Remember master password (in encrypted form) of a database while is open" is thicked or unthicked in the Security tab of the options window?

**Comment by KittQB** [comment]
Hey,
Yes that option is ticked.

**Comment by Angelelz** [comment]
Right now I'm unable to reproduce, so could you kindly provide this information:
- Masterkey components: (password, keyfile, windows user, etc?)
- Windows version
- Windows hello hardware: (Fingerprint, IR camera, or just pin)
- Anything else you feel relevant

I apologise for the inconvenience.

**Comment by KittQB** [comment]
Masterkey components: password
Windows version: Windows 10 enterprise x64
Windows hello hardware: Fingerprint

**Comment by navporky** [comment]
Hi,

[winhello_exception.txt](https://github.com/Angelelz/WinHelloUnlock/files/4675242/winhello_exception.txt)

I got this error as well. Though I am using PIN, it may be irrelevant as the key thing I noticed was my auto lock on no keypass activity was set to too short a time before the first time configuration can be completed AND in the meantime the database gets locked. 

I resolved this by disabling/ increasing the timeout for this. Hope this helps.
